src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverterFactory.cs

```diff
@@ -18,23 +19,48 @@ public override bool CanConvert(Type type)
+        }
+
+        [SuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.",
+            Justification = "The constructor has been annotated with RequiredDynamicCodeAttribute.")]
         public override JsonConverter CreateConverter(Type type, JsonSerializerOptions options)
         {
             Debug.Assert(CanConvert(type));
             return Create(type, EnumConverterOptions.AllowNumbers, namingPolicy: null, options);
         }
 
-        internal static JsonConverter Create(Type enumType, EnumConverterOptions converterOptions, JsonNamingPolicy? namingPolicy, JsonSerializerOptions options)
+        public static JsonConverter<T> Create<T>(EnumConverterOptions converterOptions, JsonSerializerOptions options, JsonNamingPolicy? namingPolicy = null)
+            where T : struct, Enum
         {
-            return (JsonConverter)Activator.CreateInstance(
-                GetEnumConverterType(enumType),
-                new object?[] { converterOptions, namingPolicy, options })!;
+            if (!IsSupportedTypeCode(Type.GetTypeCode(typeof(T))))
+            {
+                // Char-backed enums are valid in IL and F# but are not supported by System.Text.Json.
+                return new UnsupportedTypeConverter<T>();
+            }
+
+            return new EnumConverter<T>(converterOptions, namingPolicy, options);
         }
 
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
             Justification = "'EnumConverter<T> where T : struct' implies 'T : new()', so the trimmer is warning calling MakeGenericType here because enumType's constructors are not annotated. " +
             "But EnumConverter doesn't call new T(), so this is safe.")]
-        [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)]
-        private static Type GetEnumConverterType(Type enumType) => typeof(EnumConverter<>).MakeGenericType(enumType);
+        public static JsonConverter Create(Type enumType, EnumConverterOptions converterOptions, JsonNamingPolicy? namingPolicy, JsonSerializerOptions options)
+        {
+            if (!IsSupportedTypeCode(Type.GetTypeCode(enumType)))
+            {
+                // Char-backed enums are valid in IL and F# but are not supported by System.Text.Json.
+                return UnsupportedTypeConverterFactory.CreateUnsupportedConverterForType(enumType);
+            }
+
+            Type converterType = typeof(EnumConverter<>).MakeGenericType(enumType);
+            return (JsonConverter)converterType.CreateInstanceNoWrapExceptions(
+                parameterTypes: [typeof(EnumConverterOptions), typeof(JsonNamingPolicy), typeof(JsonSerializerOptions)],
+                parameters: [converterOptions, namingPolicy, options])!;
```

Comment:
It probably isn't, but the purpose of this helpers is to avoid wrapping exceptions from the constructor in `TargetInvocationException` since it can throw validation exceptions.